gui/default/syncthing/core/syncthingController.js

```diff
@@ -1729,11 +1748,35 @@ angular.module('syncthing.core')
             $('#needed').modal().on('hidden.bs.modal', function () {
                 $scope.neededFolder = undefined;
                 $scope.needed = undefined;
-                $scope.neededTotal = 0;
                 $scope.neededCurrentPage = 1;
             });
         };
 
+        $scope.showRemoteNeed = function (device) {
+            var promises = [];
+            var i = 0;
+            $scope.remoteNeedDevice = device;
+            $scope.deviceFolders(device).forEach(function(folder) {
+                if ($scope.completion[device.deviceID][folder].needCount === 0) {
+                    return;
+                }
+                $scope.remoteNeedFolders.push(folder);
+                promises.push(getRemoteNeedPromise(1, 10, i));
+                i++;
+            });
+            $q.all(promises).then(function (results) {
+                results.forEach(function(res) {
+                    $scope.remoteNeed.push(res.data);
+                });
+                $('#remoteNeed').modal().on('hidden.bs.modal', function () {
```

Comment:
How long does it take to load the remote need thing for large databases or slow devices? Given we don't show the modal until everything has resolved this might leave the user clicking then having absolutely nothing happen for seconds and seconds?

Reply:
Could be a moment indeed - now the modal is shown immediately displaying a simple "loading..." string and data as soon as the rest call for the first folder returns.